fossas/srclib-java
# [Do Not Merge] Updates log4j version to 2.16.0

## Overview

Recap from [slack](https://teamfossa.slack.com/archives/C0155DTGWB1/p1639182747276700?thread_ts=1639179273.261300&cid=C0155DTGWB1) per connor's comment:
```
- version 1.x of log4j is affected by the recent 0day if and only if the JMS appender is enabled
- srclib-java does not use the JMS appender
- srclib-java is safe
```

This PR, 

- replaces slf4j binding per, https://logging.apache.org/log4j/2.x/log4j-slf4j-impl/ to use log4j2. 
- uses log4j version 2.16.0

## Test

1. Checkout branch
2. Run `fossa analyze -o | jq | grep "log4j"`

Notice, all log4k versions are 2.16.0 or greater:
```
➜  srclib-java git:(fix/uses-log4j2-for-cve-2021-44228) ✗ fossa analyze -o | jq | grep "log4j"              
[ INFO] [ 0 Waiting / 1 Running / 0 Completed ]
[ INFO] [ 22 Waiting / 12 Running / 1 Completed ]
[ INFO] [ 11 Waiting / 12 Running / 12 Completed ]
[ INFO] [ 5 Waiting / 12 Running / 18 Completed ]
[ INFO] [ 4 Waiting / 12 Running / 19 Completed ]
[ INFO] [ 0 Waiting / 12 Running / 23 Completed ]
[ INFO] [ 0 Waiting / 11 Running / 24 Completed ]
[ INFO] [ 0 Waiting / 10 Running / 25 Completed ]
[ INFO] [ 0 Waiting / 9 Running / 26 Completed ]
[ INFO] [ 0 Waiting / 8 Running / 27 Completed ]
[ INFO] [ 0 Waiting / 7 Running / 28 Completed ]
[ INFO] [ 0 Waiting / 6 Running / 29 Completed ]
[ INFO] [ 0 Waiting / 5 Running / 30 Completed ]
[ INFO] [ 0 Waiting / 4 Running / 31 Completed ]
[ INFO] [ 0 Waiting / 3 Running / 33 Completed ]
[ INFO] Analyzing maven project at /Users/megh/Work/upstream/srclib-java/src/test/resources/repo/foo/bar/dummy-lib/1.0/
[ INFO] [ 0 Waiting / 2 Running / 34 Completed ]
[ INFO] Analyzing gradle project at /Users/megh/Work/upstream/srclib-java/
[ INFO] [ 0 Waiting / 2 Running / 35 Completed ]
[ INFO] [ 0 Waiting / 1 Running / 36 Completed ]

            "name": "org.apache.logging.log4j:log4j-api",
            "name": "org.apache.logging.log4j:log4j-core",
            "name": "org.apache.logging.log4j:log4j-slf4j-impl",
            "locator": "mvn+org.apache.logging.log4j:log4j-api$2.16.0"
              "mvn+org.apache.logging.log4j:log4j-api$2.16.0"
            "locator": "mvn+org.apache.logging.log4j:log4j-core$2.16.0"
              "mvn+org.apache.logging.log4j:log4j-api$2.16.0",
              "mvn+org.apache.logging.log4j:log4j-core$2.16.0",
            "locator": "mvn+org.apache.logging.log4j:log4j-slf4j-impl$2.16.0"
          "mvn+org.apache.logging.log4j:log4j-slf4j-impl$2.16.0",
                    
➜  srclib-java git:(fix/uses-log4j2-for-cve-2021-44228) ✗ 
```
